modules/aggs-matrix-stats/src/test/java/org/elasticsearch/search/aggregations/matrix/stats/InternalMatrixStatsTests.java

```diff
@@ -100,4 +127,55 @@ public void testReduceRandom() {
     protected void assertReduced(InternalMatrixStats reduced, List<InternalMatrixStats> inputs) {
         throw new UnsupportedOperationException();
     }
+
+    @Override
+    protected void assertFromXContent(InternalMatrixStats expected, ParsedAggregation parsedAggregation) throws IOException {
+        assertTrue(parsedAggregation instanceof ParsedMatrixStats);
+        ParsedMatrixStats actual = (ParsedMatrixStats) parsedAggregation;
+
+        for (String field : fields) {
+            assertEquals(expected.getFieldCount(field), actual.getFieldCount(field));
+            assertEquals(expected.getMean(field), actual.getMean(field), 0.0);
+            assertEquals(expected.getVariance(field), actual.getVariance(field), 0.0);
+            assertEquals(expected.getSkewness(field), actual.getSkewness(field), 0.0);
+            assertEquals(expected.getKurtosis(field), actual.getKurtosis(field), 0.0);
+
+            for (String other : fields) {
+                assertEquals(expected.getCovariance(field, other), actual.getCovariance(field, other), 0.0);
+                assertEquals(expected.getCorrelation(field, other), actual.getCorrelation(field, other), 0.0);
+            }
+        }
+
+        final String unknownField = randomAlphaOfLength(3);
+        final String fieldX = randomFrom(unknownField, randomAlphaOfLength(3));
```

Comment:
Yeah, it's not really readable. I changed that. The idea is to test getCovariance/getCorrelation with various unknown fields.